parser/src/aspects/substitution.rs

```diff
@@ -14,32 +16,36 @@ pub(crate) trait SubstitutionAspect<'a> {
 
 impl<'a> SubstitutionAspect<'a> for Parser<'a> {
     fn substitution(&mut self) -> ParseResult<Expr<'a>> {
-        let start_token = self.cursor.force(
-            of_types(&[TokenType::At, TokenType::Dollar]),
-            "Expected at or dollar sign.",
-        )?;
+        let dollar_value = self.cursor.force(
+            of_type(TokenType::Dollar),
+            "Expected '$' sign.",
+        )?.value;
+
+        //if $ is directly followed by a '(' then it's the start of a Capture substitution.
+        if self.cursor
+            .lookahead(of_type(RoundedLeftBracket))
+            .is_some() {
+            return Ok(Expr::Substitution(Substitution {
+                // Read the expression inside the parentheses as a new statement
+                underlying: self.subshell()?,
+                kind: SubstitutionKind::Capture,
+            }))
+        }
 
         // Short pass for variable references
-        if self
-            .cursor
-            .lookahead(of_type(TokenType::RoundedLeftBracket))
-            .is_none()
-            && start_token.token_type == TokenType::Dollar
+        if self.cursor
+            .lookahead(
+                like(TokenType::is_valid_var_ref_name)
```

Comment:
The lookahead shouldn't check too early that the variable name is valid. This cause that invalid variable names to be ignored by the `var_reference` parser, which would raise a proper and appropriate error for it.